### DISCUSSION TITLE
Developing an extension library to add the request aggregation functionallity to the YARP.

### DISCUSSION THREAD

**Discussion by shokri-navid** [dotnet/yarp#2633] [Ideas] [answered]
I am looking for the functionality of request aggregation inside YARP and [this thread of GitHub issue](https://github.com/microsoft/reverse-proxy/issues/1414) caught my eye. 

I see @samsp-msft's point. However, request aggregation could be an extension of the YARP. Is there a solution to add this functionality as a separate extension library that leaves the core untouched but makes programmers' lives easier?
Based on my experience performance is a MUST aspect for API Gateways / Reverse Proxies and it is the main motivation for engineers to use YARP. But, the Request aggregation is a common requirement these days especially when you are working with legacy code.
I want to check the possibility of this approach and some key hints to estimate that could I develop such an extension.

**Comment by MihaZupan** [comment, marked as answer]
See https://github.com/microsoft/reverse-proxy/discussions/1677#discussioncomment-5104010, https://github.com/microsoft/reverse-proxy/issues/2386#issuecomment-1966183150.

It's certainly possible to build a separate library that builds on top of YARP to add more functionality, but it's unclear what YARP features you would want to reuse/build on.

If the logic around how you combine responses together is always the same, there were examples posted in https://github.com/microsoft/reverse-proxy/discussions/1677 with code showing how that could be done.
